src/main/kotlin/dev/bopke/celestIslesSkyblock/island/IslandRepository.kt

```diff
@@ -0,0 +1,18 @@
+package dev.bopke.celestIslesSkyblock.island
+
+import com.dzikoysk.sqiffy.dsl.Column
+import org.bukkit.World
+import java.util.*
+import java.util.concurrent.CompletableFuture
+
+interface IslandRepository {
+
+    fun create(creatorUuid: UUID, world: World, islandName: String): UnidentifiedIsland
+
+    fun insert(island: UnidentifiedIsland): CompletableFuture<Island>
+
+    fun <T> update(creatorUuid: UUID, column: Column<T>, value: T): CompletableFuture<Int>
```

Comment:
maybe instead of insert and update methods, lets go with save() approach which should be overloaded, in case it is provided UnidentifiedIsland it will call insert, and in case of Island it would call update for all fields at once. Doing it column by column would be cumbersome and easy to forget some fields, and also creatorUuid is not proper ID to identify single Island